### PR TITLE
API-10157: InsurancePlan controller support for vista RPC

### PR DESCRIPTION
**Pull request by jabando-va**

### CODE REVIEW

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/LhsGatewayErrorHandler.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ private String unknownRecordVistaCoordinates() {
   }
 
   /** Verify there are errors and deal with them appropriately. */
-  void validateResults() {
+  public void validateResults() {
```
[the remark this answers is not in the document]
Comment (jabando-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/LhsGatewayErrorHandler.java:54]: To this?
```
package gov.va.api.health.vistafhirquery.service.controller

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4InsurancePlanToGroupInsurancePlanFileTransformer.java [hunk 2]
```diff
@@ -225,7 +225,7 @@ WriteableFilemanValue typeOfPlan(List<InsurancePlan.Plan> plan) {
         c ->
             codeableconceptHasCodingSystem(
                 c.type(), "urn:oid:2.16.840.1.113883.3.8901.3.1.355803.8009"),
-        c -> extractCodeFromCoding(c.type()),
+        c -> c.type().coding().stream().findFirst().map(Coding::display).orElse(""),
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4InsurancePlanToGroupInsurancePlanFileTransformer.java:228]: Why would we not get the coding for the correct type?

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4SiteInsurancePlanController.java [hunk 3]
```diff
@@ -100,11 +120,23 @@ public InsurancePlan insurancePlanRead(
     var response = charon.request(request);
     var lhsResponse = lighthouseRpcGatewayResponse(response);
     dieOnError(lhsResponse);
-
     var resources = transformation().toResource().apply(lhsResponse);
     return verifyAndGetResult(resources, id);
   }
 
+  private LhsLighthouseRpcGatewayCoverageWrite.Request organizationWriteDetails(
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4SiteInsurancePlanController.java:127]: Copy Pasta

Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4SiteInsurancePlanController.java:127]: Not sure it's worth it, but something to think about. All write mappings use almost this exact method.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4InsurancePlanToGroupInsurancePlanFileTransformer.java [hunk 4]
```diff
@@ -225,7 +225,7 @@ WriteableFilemanValue typeOfPlan(List<InsurancePlan.Plan> plan) {
         c ->
             codeableconceptHasCodingSystem(
                 c.type(), "urn:oid:2.16.840.1.113883.3.8901.3.1.355803.8009"),
-        c -> extractCodeFromCoding(c.type()),
+        c -> c.type().coding().stream().findFirst().map(Coding::display).get(),
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4InsurancePlanToGroupInsurancePlanFileTransformer.java:228]: What if the display isn't present?

**Review by jhulbert-va — APPROVED** [review]

**Pull request merged**